src/components/auth/registerUser.functional.spec.js

```diff
@@ -0,0 +1,139 @@
+import signUpModule from './signup/signup';
+import signUpCtrl from './signup/signup.controller';
+import signUpTemplate from './signup/signup.html';
+import directives from '../../directives/email';
+import app from './../../index.js';
+
+describe('register user functional test', function() {
+  let $rootScope;
+  let makeController;
+  let makeDirective;
+  let makeTemplate
+  let $toast;
+  let $state;
+  let $auth;
+  let $compile;
+  let form;
+  let AuthService;
+  let $httpBackend;
+
+  beforeEach(angular.mock.module('app'));
+
+  beforeEach(inject((_$rootScope_, _$toast_, _$state_, _$auth_, _$compile_, _AuthService_, _$httpBackend_) => {
+    $rootScope = _$rootScope_;
+
+    $httpBackend = _$httpBackend_;
+
+    $toast = _$toast_;
+
+    $state = _$state_;
+
+    $auth = _$auth_;
+
+    $compile = _$compile_;
+
+    AuthService = _AuthService_;
+
+    makeTemplate = angular.element(signUpTemplate);
+
+    $compile(makeTemplate)($rootScope);
```

Comment:
You should compile the sign-up template into sign-up controller instead of $rootScope.